### PR TITLE
Buildable mass drivers

### PR DESCRIPTION
**Pull request by Vincent983**
## About The Pull Request
Adds the ability to build mass drivers, you activate them by attaching a signaler to their launch wire. They also have a safety wire, that when pulsed, increases the power of it. To reset the power back to normal you cut the safety wire. This is mostly just a port of https://github.com/BeeStation/BeeStation-Hornet/pull/5563

## Why It's Good For The Game

Less weird unique machines, and you can do some interesting stuff with mass drivers, like traps, cannons, and many other interesting things.

## Changelog

:cl:
add: mass drivers are now buildable, you activate them by attaching a signaler to their launch wire, and can increase their power by pulsing the safeties wire, and reset it back to normal by cutting the safeties wire.
/:cl:

### CODE REVIEW

**Comment by Vincent983** [comment]
fuck, where the hell is the space indentation

File: code/datums/wires/mass_driver.dm [hunk 1]
```diff
@@ -0,0 +1,25 @@
+/datum/wires/mass_driver
+    holder_type = /obj/machinery/mass_driver
+    proper_name = "Mass Driver"
+
+/datum/wires/mass_driver/New(atom/holder)
+    wires = list(WIRE_LAUNCH, WIRE_SAFETIES)
+    ..()
+
+/datum/wires/mass_driver/on_pulse(wire)
+    var/obj/machinery/mass_driver/M = holder
```
Comment (necromanceranne) [inline code/datums/wires/mass_driver.dm:10]: In any instance where you have a variable name that uses a single letter or abbreviation, you should use a more descriptive name.

For this, we could instead use
```suggestion
    var/obj/machinery/mass_driver/our_mass_driver = holder
```
This is a part of the style guide we use.  https://github.com/tgstation/tgstation/blob/master/.github/guides/STYLE.md#variables

**Comment by ju45he** [comment]
Can 40 glass shards instantly kill a death squad member? Today we'll find out on greytiders united.

File: code/datums/wires/mass_driver.dm (hunk 1 above)
Comment (Vincent983) [reply code/datums/wires/mass_driver.dm:10]: alright

**Comment by Vincent983** [comment]
> Can 40 glass shards instantly kill a death squad member? Today we'll find out on greytiders united.

no idea, but it is hilarious to put 5 spears/cult daggers in a hacked mass driver, and watch a man get impaled

**Comment by CCC23** [comment]
Time to strap a maxcap production station on a whiteship and mass driver bombs at people

**Comment by CRITAWAKETS** [comment]
We have the musket, we have the smoothbore

and now we have the cannon. TALLY-HO, LADS!

**Review by Wallemations — APPROVED** [review]
Woah I wonder who made that sprite

**Pull request merged**